src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -271,17 +242,16 @@ public class UnsolvedSymbolVisitor extends ModifierVisitor<Void> {
   public UnsolvedSymbolVisitor(
       String rootDirectory,
       Map<String, Path> existingClassesToFilePath,
-      List<String> targetMethodsSignatures,
-      List<String> targetFieldsSignature) {
+      Set<String> targetMethodsSignatures,
```

Comment:
We may want to rename this to `targetMethodNames` since this set contains user-provided signatures like `com.example.Class#method()` rather than the typical `getQualifiedSignature()` signatures like `com.example.Class.method()`.

Reply:
We could also just change formats here (replace `#` with `.`), since this set does get passed down to other visitors which use `getQualifiedSignature()`.

Reply:
> We may want to rename this to targetMethodNames since this set contains user-provided signatures like com.example.Class#method() rather than the typical getQualifiedSignature() signatures like com.example.Class.method().

I don't think that this is correct: we never do such a conversion as far as I know (and there are many comparisons between the result of a call to `getQualifiedSignature` and the elements of `SpeciminStateVisitor#targetMethods`, which are the same as the contents of this set with the spaces removed.

Reply:
Yep, I agree. However, a conversion (not really) occurs in `TargetMethodFinderVisitor` [here](https://github.com/njit-jerse/specimin/blob/9ea46fe9dd6cf55a2405680c14e02a9927f43428/src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java#L274) so I believe the `targetMethods` set (since it is never cleared) contains both the user-specified signatures and `getQualifiedSignature` signatures past that visitor. This doesn't affect `UnsolvedSymbolVisitor` so it doesn't really matter. Sorry!

Reply:
> a conversion (not really) occurs in TargetMethodFinderVisitor [here](https://github.com/njit-jerse/specimin/blob/9ea46fe9dd6cf55a2405680c14e02a9927f43428/src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java#L274) so I believe the targetMethods set (since it is never cleared) contains both the user-specified signatures and getQualifiedSignature signatures past that visitor

I'll open another PR maybe to address this duplication - I don't think it needs to be mixed with this already large one. Thanks for pointing this out.

Reply:
I looked into doing this, and it creates more problems than it solves. In particular, it's difficult to standardize on one internal format for the target methods set for three, competing reasons:
* `UnsolvedSymbolVisitor` needs to run before it is possible to resolve methods
* method signatures need to have fully-qualified parameter types to match the output of `ResolvedMethodDeclaration#getQualifiedSignature`
* we don't want to force users to fully qualify parameter types in the input.

Put another way, there are three main differences between the "getQualifiedSignature" format and the current input format:
* the current input format uses a `#` between the class and method name, but `getQualifiedSignature` uses a `.`
* the input format permits internal whitespace
* the input format permits simple names for parameter types

The first two of these differences can be overcome with string manipulation, but the last cannot. Maybe there is a better refactoring here, but I don't immediately see it - maybe creating a special class for "method signatures" that stores both the input format signature and the qualified signature (once it's available)? But that seems overly complex to me.